TypeCobol/Compiler/Parser/CodeElementsParserStep.cs

```diff
@@ -187,13 +187,41 @@ internal static IList<DocumentChange<ICodeElementsLine>> ParseProcessedTokensLin
                 {
                     tokenStream.StartLookingForStopToken(largestRefreshParseSection.StopToken);
 
-                    //Remove all the code elements for the future line to parse.
+                    // Get active REPLACE directive from line located right before reparse section
+                    ReplaceDirective activeReplaceDirective;
+                    if (largestRefreshParseSection.StartLineIndex > 0)
+                    {
+                        var lineBefore = documentLines[largestRefreshParseSection.StartLineIndex - 1];
+                        if (lineBefore.ImportedDocuments != null)
+                        {
+                            // The preceding line imports copies, use last line of the last resolved copy
+                            var lastLine = lineBefore.ImportedDocuments.Values
+                                .Where(d => d != null)
+                                .SelectMany(d => d.SourceDocument.Lines)
+                                .LastOrDefault();
```

Comment:
For performance reason we should avoid SelectMany.
```suggestion
                            var lastLine = lineBefore.ImportedDocuments.Values.Last(d=>d!=null).SourceDocument.Lines.LastOrDefault();
```

Reply:
Reverted.
As long as the CodeElementsParserStep reparses all lines affected by a modified REPLACE, using the value of `ActiveReplaceDirective` coming from a previous parsing round should be fine. So I suggest to keep the previous code.